### PR TITLE
Convert ColorUtilTest to Kotlin

### PR DESCRIPTION
**Pull request by okwasniewski**
## Summary:

Converts ColorUtilTest to kotlin as requested in #38825

## Changelog:

[INTERNAL] [CHANGED] - Convert ColorUtilTest to Kotlin

## Test Plan:

1. Run `./gradlew :packages:react-native:ReactAndroid:test`.
2. All tests should pass.

### CODE REVIEW

File: packages/react-native/ReactAndroid/src/test/java/com/facebook/react/views/view/ColorUtilTest.kt [hunk 1]
```diff
@@ -0,0 +1,80 @@
...
+ */
+package com.facebook.react.views.view
+
+import android.graphics.PixelFormat
+import org.junit.Assert
+import org.junit.Rule
+import org.junit.Test
+import org.junit.runner.RunWith
+import org.powermock.core.classloader.annotations.PowerMockIgnore
+import org.powermock.modules.junit4.rule.PowerMockRule
```
Comment (cortinico) [inline packages/react-native/ReactAndroid/src/test/java/com/facebook/react/views/view/ColorUtilTest.kt:15]:
```suggestion
```

File: packages/react-native/ReactAndroid/src/test/java/com/facebook/react/views/view/ColorUtilTest.kt [hunk 2]
```diff
@@ -0,0 +1,80 @@
...
+import org.robolectric.RobolectricTestRunner
+
+/** Based on Fresco's DrawableUtilsTest (https://github.com/facebook/fresco). */
+@RunWith(RobolectricTestRunner::class)
+@PowerMockIgnore("org.mockito.*", "org.robolectric.*", "androidx.*", "android.*")
+class ColorUtilTest {
+  @get:Rule var rule = PowerMockRule()
```
Comment (cortinico) [inline packages/react-native/ReactAndroid/src/test/java/com/facebook/react/views/view/ColorUtilTest.kt:22]:
```suggestion
```

File: packages/react-native/ReactAndroid/src/test/java/com/facebook/react/views/view/ColorUtilTest.kt [hunk 3]
```diff
@@ -0,0 +1,80 @@
...
+import org.junit.Rule
+import org.junit.Test
+import org.junit.runner.RunWith
+import org.powermock.core.classloader.annotations.PowerMockIgnore
+import org.powermock.modules.junit4.rule.PowerMockRule
+import org.robolectric.RobolectricTestRunner
+
+/** Based on Fresco's DrawableUtilsTest (https://github.com/facebook/fresco). */
+@RunWith(RobolectricTestRunner::class)
+@PowerMockIgnore("org.mockito.*", "org.robolectric.*", "androidx.*", "android.*")
```
Comment (cortinico) [inline packages/react-native/ReactAndroid/src/test/java/com/facebook/react/views/view/ColorUtilTest.kt:20]:
```suggestion
```

File: packages/react-native/ReactAndroid/src/test/java/com/facebook/react/views/view/ColorUtilTest.kt [hunk 4]
```diff
@@ -0,0 +1,80 @@
...
+class ColorUtilTest {
+  @get:Rule var rule = PowerMockRule()
+
+  @Test
+  fun testMultiplyColorAlpha() {
+    Assert.assertEquals(0x00123456, ColorUtil.multiplyColorAlpha(-0x3fedcbaa, 0).toLong())
+    Assert.assertEquals(0x07123456, ColorUtil.multiplyColorAlpha(-0x3fedcbaa, 10).toLong())
+    Assert.assertEquals(-0x69edcbaa, ColorUtil.multiplyColorAlpha(-0x3fedcbaa, 200).toLong())
+    Assert.assertEquals(-0x3fedcbaa, ColorUtil.multiplyColorAlpha(-0x3fedcbaa, 255).toLong())
```
Comment (cortinico) [inline packages/react-native/ReactAndroid/src/test/java/com/facebook/react/views/view/ColorUtilTest.kt:29]: Can we avoid using the `-` and retain the old values?

File: packages/react-native/ReactAndroid/src/test/java/com/facebook/react/views/view/ColorUtilTest.kt [hunk 5]
```diff
@@ -0,0 +1,80 @@
...
+      PixelFormat.TRANSPARENT.toLong(),
+      ColorUtil.getOpacityFromColor(0x00123456).toLong()
+    )
+    Assert.assertEquals(
+      PixelFormat.TRANSPARENT.toLong(),
+      ColorUtil.getOpacityFromColor(0x00FFFFFF).toLong()
+    )
+    Assert.assertEquals(
+      PixelFormat.TRANSLUCENT.toLong(),
+      ColorUtil.getOpacityFromColor(-0x40000000).toLong()
```
Comment (cortinico) [inline packages/react-native/ReactAndroid/src/test/java/com/facebook/react/views/view/ColorUtilTest.kt:48]: same as above

File: packages/react-native/ReactAndroid/src/test/java/com/facebook/react/views/view/ColorUtilTest.kt [hunk 6]
```diff
@@ -0,0 +1,80 @@
...
+    )
+    Assert.assertEquals(
+      PixelFormat.OPAQUE.toLong(),
+      ColorUtil.getOpacityFromColor(-0x1).toLong()
+    )
+  }
+
+  @Test
+  fun testNormalize() {
+    Assert.assertEquals(-0x7ff4e9df, ColorUtil.normalize(11.0, 22.0, 33.0, 0.5).toLong())
```
Comment (cortinico) [inline packages/react-native/ReactAndroid/src/test/java/com/facebook/react/views/view/ColorUtilTest.kt:74]: same as above

**Review by cortinico — CHANGES_REQUESTED** [review]
Thanks for sending this over @okwasniewski!
Can I ask you to follow-up on my comments and then we can import/merge this

**Comment by okwasniewski** [comment]
Hey @cortinico thanks for the review. I've applied your suggestions.

**Comment by cortinico** [comment]
Converted to draft, please move it back once it's ready for review.


The tests that are failing are:
```
com.facebook.react.views.view.ColorUtilTest > testNormalize FAILED
    junit.framework.AssertionFailedError at ColorUtilTest.kt:43

com.facebook.react.views.view.ColorUtilTest > testMultiplyColorAlpha FAILED
    junit.framework.AssertionFailedError at ColorUtilTest.kt:24
```

**Comment by okwasniewski** [comment]
@cortinico Tests are now passing - ready for rereview

**Review by mdvacca — APPROVED** [review]
lgtm

**Pull request closed without merging**